Tests fail when building ES

I've checked out ES source with v1.5.0 tag and when I try to build tests fail with the following tests:
* org.elasticsearch.bwcompat.RestoreBackwardsCompatTests.restoreOldSnapshots
* org.elasticsearch.bwcompat.OldIndexBackwardsCompatibilityTests.testAllVersionsTested

This happens on Mac OS X with Java 8 + Maven 3.2.5

The README advises to skip tests, but it's bad practice to routinely do this since this way contributors won't be able to get any confirmation that their code actually works apart from it being syntactically correct.

```
mvn clean package
...
Tests with failures:
  - org.elasticsearch.bwcompat.RestoreBackwardsCompatTests.restoreOldSnapshots
  - org.elasticsearch.bwcompat.OldIndexBackwardsCompatibilityTests.testAllVersionsTested


[INFO] JVM J0:     0.67 ..   541.62 =   540.95s
[INFO] JVM J1:     0.67 ..   541.47 =   540.80s
[INFO] JVM J2:     0.67 ..   558.75 =   558.08s
[INFO] JVM J3:     0.67 ..   541.54 =   540.87s
[INFO] Execution time total: 9 minutes 18 seconds
[INFO] Tests summary: 708 suites, 4699 tests, 2 failures, 70 ignored (59 assumptions)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 09:55 min
[INFO] Finished at: 2015-03-27T10:26:36+02:00
[INFO] Final Memory: 31M/1331M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.carrotsearch.randomizedtesting:junit4-maven-plugin:2.1.2:junit4 (tests) on project elasticsearch: Execution tests of goal com.carrotsearch.randomizedtesting:junit4-maven-plugin:2.1.2:junit4 failed: /Users/marko/projects/3rd-party/elasticsearch/target/junit4-ant-7275726970577951678.xml:16: There were test failures: 708 suites, 4699 tests, 2 failures, 70 ignored (59 assumptions) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
~/projects/3rd-party/elasticsearch (5448160 ✔) ᐅ java -version
java version "1.8.0_40"
Java(TM) SE Runtime Environment (build 1.8.0_40-b27)
Java HotSpot(TM) 64-Bit Server VM (build 25.40-b25, mixed mode)```

--- a/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java
+++ b/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java
@@ -65,9 +65,9 @@ import static org.hamcrest.CoreMatchers.equalTo;
 public class OldIndexBackwardsCompatibilityTests extends StaticIndexBackwardCompatibilityTest {
     // TODO: test for proper exception on unsupported indexes (maybe via separate test?)
     // We have a 0.20.6.zip etc for this.
-    
+
     static List<String> indexes;
-    
+
     @BeforeClass
     public static void initIndexes() throws Exception {
         indexes = new ArrayList<>();
@@ -80,19 +80,20 @@ public class OldIndexBackwardsCompatibilityTests extends StaticIndexBackwardComp
         }
         Collections.sort(indexes);
     }
-    
+
     public void testAllVersionsTested() throws Exception {
         SortedSet<String> expectedVersions = new TreeSet<>();
         for (java.lang.reflect.Field field : Version.class.getDeclaredFields()) {
             if (Modifier.isStatic(field.getModifiers()) && field.getType() == Version.class) {
                 Version v = (Version)field.get(Version.class);
-                if (v.snapshot()) continue;
-                if (v.onOrBefore(Version.V_0_20_6)) continue;
+                if (v.snapshot()) continue;  // snapshots are unreleased, so there is no backcompat yet
+                if (v.onOrBefore(Version.V_0_20_6)) continue; // we can only test back one major lucene version
+                if (v.equals(Version.CURRENT)) continue; // the current version is always compatible with itself
 
                 expectedVersions.add("index-" + v.toString() + ".zip");
             }
         }
-        
+
         for (String index : indexes) {
             if (expectedVersions.remove(index) == false) {
                 logger.warn("Old indexes tests contain extra index: " + index);
@@ -131,11 +132,11 @@ public class OldIndexBackwardsCompatibilityTests extends StaticIndexBackwardComp
         assertDeleteByQueryWorked(version);
         unloadIndex();
     }
-    
+
     Version extractVersion(String index) {
         return Version.fromString(index.substring(index.indexOf('-') + 1, index.lastIndexOf('.')));
     }
-    
+
     boolean isLatestLuceneVersion(Version version) {
         return version.luceneVersion.major == Version.CURRENT.luceneVersion.major &&
                version.luceneVersion.minor == Version.CURRENT.luceneVersion.minor;
@@ -147,7 +148,7 @@ public class OldIndexBackwardsCompatibilityTests extends StaticIndexBackwardComp
         ElasticsearchAssertions.assertNoFailures(searchRsp);
         long numDocs = searchRsp.getHits().getTotalHits();
         logger.info("Found " + numDocs + " in old index");
-        
+
         searchReq.addSort("long_sort", SortOrder.ASC);
         ElasticsearchAssertions.assertNoFailures(searchReq.get());
 
@@ -229,7 +230,7 @@ public class OldIndexBackwardsCompatibilityTests extends StaticIndexBackwardComp
         SearchRequestBuilder searchReq = client().prepareSearch("test").setQuery(QueryBuilders.queryStringQuery("long_sort:[10 TO 20]"));
         assertEquals(0, searchReq.get().getHits().getTotalHits());
     }
-    
+
     void assertUpgradeWorks(boolean alreadyLatest) throws Exception {
         HttpRequestBuilder httpClient = httpClient();
 

--- a/src/test/java/org/elasticsearch/bwcompat/RestoreBackwardsCompatTests.java
+++ b/src/test/java/org/elasticsearch/bwcompat/RestoreBackwardsCompatTests.java
@@ -72,6 +72,7 @@ public class RestoreBackwardsCompatTests extends AbstractSnapshotTests {
                 Version v = (Version) field.get(Version.class);
                 if (v.snapshot()) continue;
                 if (v.onOrBefore(Version.V_1_0_0_Beta1)) continue;
+                if (v.equals(Version.CURRENT)) continue;
 
                 expectedVersions.add(v.toString());
             }

